src/test/java/org/apache/accumulo/access/AccessExpressionBenchmark.java

```diff
@@ -0,0 +1,183 @@
+                            expressions.add(byteExp);
+                            et.expressions.add(byteExp);
+                            et.parsedExpressions.add(AccessExpression.of(exp));
+                        }
+                    }
+                }
+
+                evaluatorTests.add(et);
+            }
+        }
+
+        List<byte[]> getBytesExpressions(){
+            return expressions;
+        }
+
+        List<String> getStringExpressions(){
+            return strExpressions;
+        }
+
+        public ArrayList<EvaluatorTests> getEvaluatorTests() {
+            return evaluatorTests;
+        }
+
+    }
+
+    /**
+     * Measures the time it takes to parse an expression stored in byte[] and produce a parse tree.
+     */
+    @Benchmark
+    public void measureBytesParsing(BenchmarkState state, Blackhole blackhole) {
+        for(byte[] accessExpression : state.getBytesExpressions()) {
+            blackhole.consume(AccessExpression.of(accessExpression));
+        }
+    }
+
+    /**
+     * Measures the time it takes to parse an expression stored in a String and produce a parse tree.
+     */
+    @Benchmark
+    public void measureStringParsing(BenchmarkState state, Blackhole blackhole) {
+        for(String accessExpression : state.getStringExpressions()) {
+            blackhole.consume(AccessExpression.of(accessExpression));
+        }
+    }
+
+    /**
+     * Measures the time it takes to evaluate a previously parsed expression.
+     */
+    @Benchmark
+    public void measureEvaluation(BenchmarkState state, Blackhole blackhole) {
+        for(EvaluatorTests evaluatorTests : state.getEvaluatorTests()) {
+            for(AccessExpression expression : evaluatorTests.parsedExpressions) {
+                blackhole.consume(evaluatorTests.evaluator.canAccess(expression));
+            }
+        }
+    }
+
+    /**
+     * Measures the time it takes to parse and evaluate an expression.  This has to create the parse tree an operate on it.
+     */
+    @Benchmark
+    public void measureEvaluationAndParsing(BenchmarkState state, Blackhole blackhole) {
+        for(EvaluatorTests evaluatorTests : state.getEvaluatorTests()) {
+            for(byte[] expression : evaluatorTests.expressions) {
+                blackhole.consume(evaluatorTests.evaluator.canAccess(expression));
+            }
+        }
+    }
+
+    public static void main(String[] args) throws RunnerException, IOException {
+
+        var state = new BenchmarkState();
+        state.loadData();
+
+        int numExpressions =state.getBytesExpressions().size();
+
+        Options opt = new OptionsBuilder()
+                .include(AccessExpressionBenchmark.class.getSimpleName())
+                .operationsPerInvocation(numExpressions)
+                .timeUnit(TimeUnit.MICROSECONDS)
+                .warmupTime(TimeValue.seconds(5))
+                .warmupIterations(3)
+                .measurementIterations(4)
```

Comment:
I don't know if there are enough iterations here to trigger the JIT compiler. In a long running server process the `AccessExpression.of` and/or underlying methods will likely be compiled. If you want to simulate that you could either lower the CompilerThreshold or increase the iterations.

Reply:
[This](https://stackoverflow.com/questions/72482511/ensuring-jmh-benchmark-method-gets-compiled) talks about using the `jvmArgsAppend` parameter of the `Fork` annotation to modify the CompilerThreshold for the benchmark method to 5% of the default (10,000). Another option, maybe in conjunction with lowering the CompilerThreshold, would be iterate over the expressions N times in the test methods, getting over the 10,000 invocation default.

Reply:
I feel like the JIT is kicking in during the runs because it spends 15 seconds in warmup running the methods as many times as it can.  So I think it runs the methods millions of times during warmup.  I don't see things change when giving more time to warmup

Reply:
Ok, I wasn't sure how many calls had been made.

Reply:
Looking at [this](https://stackoverflow.com/questions/1503479/how-to-see-jit-compiled-code-in-jvm) I found I could add the following the benchmark builder and see when it compiles methods.  It produces a lot of output.

```
.jvmArgs("-XX:+UnlockDiagnosticVMOptions","-XX:CompileCommand=print,*access/*.*")
```

Reply:
I cloned your branch and ran the Benchmark. I think I understand these settings a little more. Here is what I'm seeing:

1. The benchmark runs each benchmark method in a forked JVM
2. The forked JVM run serially, not in parallel
3. The forked JVM runs the benchmark method 3 times as a warmup, timing out if it takes longer than 5s
4. The forked JVM runs the benchmark method 4 times and captures these measurements 
5. In total, each benchmark method is run 12 times.

Reply:
The number of operations per invocation isn't correct for all of the tests, for example the evaluations tests the number of operations is (numExpressions * numEvaluationTests). I guess as long as it's the same setup for each test, then it should be sufficient to compare two different implementations against each other. Just need to realize that for the evaluations tests, the timing information in the results may not be correct.

Reply:
> I feel like the JIT is kicking in during the runs because it spends 15 seconds in warmup running the methods as many times as it can. So I think it runs the methods millions of times during warmup. I don't see things change when giving more time to warmup

I didn't realize that it called the benchmark method continuously until the timeout.

Reply:
> The forked JVM run serially, not in parallel

@dlmarion  one thing I have noticed a few times when running is that a fork will consistently perform differently. Curious if you observed this.  Wonder if the JIT compiles code differently sometimes.

Reply:
I only noticed minor variations in the ops/us numbers, usually +/- 1.0. I was running this on my EC2 dev box and shutdown the IDE. Nothing else was running. I also ran with `taskset -c 3`.